apps/webapp/app/services/jobs/disableJob.server.ts

```diff
@@ -0,0 +1,103 @@
+import type { JobVersion } from "@trigger.dev/database";
+import type { PrismaClient } from "~/db.server";
+import { prisma } from "~/db.server";
+import { ExtendedEndpoint, findEndpoint } from "~/models/endpoint.server";
+import type { AuthenticatedEnvironment } from "../apiAuth.server";
+import { DisableScheduleSourceService } from "../schedules/disableScheduleSource.server";
+
+export type DisableJobServiceOptions = {
+  slug: string;
+  version: string;
+};
+
+export class DisableJobService {
+  #prismaClient: PrismaClient;
+
+  constructor(prismaClient: PrismaClient = prisma) {
+    this.#prismaClient = prismaClient;
+  }
+
+  public async call(
+    endpointIdOrEndpoint: string | ExtendedEndpoint,
+    options: DisableJobServiceOptions
+  ) {
+    const endpoint =
+      typeof endpointIdOrEndpoint === "string"
+        ? await findEndpoint(endpointIdOrEndpoint)
+        : endpointIdOrEndpoint;
+
+    return this.#disableJob(endpoint.environment, options);
+  }
+
+  async #disableJob(
+    environment: AuthenticatedEnvironment,
+    options: DisableJobServiceOptions
+  ): Promise<JobVersion | undefined> {
+    // Find the job
+    const job = await this.#prismaClient.job.findUnique({
+      where: {
+        projectId_slug: {
+          projectId: environment.projectId,
+          slug: options.slug,
+        },
+      },
+    });
+
+    if (!job) {
+      return;
```

Comment:
Is it worth adding a log here? Feels like this shouldn't happen

Reply:
It's just defensive programming against thrown errors, but yea I'll add a log